### PR TITLE
Add new "quickstart" samples.

### PR DESCRIPTION
**Pull request by jmdobry**
These are intended to be simple examples that explicitly show:
- Importing the client library
- Instantiating a client
- Making a single API call
- Printing the result

**Note:** This is my first attempt at writing Go, please be nice 😄 

**Note 2:** The variable names, region tags, structure of the sample, etc. were all carefully chosen and currently match the "quickstart" samples for other languages (linked to below), so this is how we'd like the samples to look. But, if I've done something glaringly non-idiomatic then let's fix that.

**Note 3:** I'd love to have provided tests, but can you (golang-samples maintainers) help out with the tests? I'm off writing these samples in 6 other languages... I ran all of the samples locally to make sure they work.

**Products covered:**
- BigQuery
  - [x] - sample, [already published](https://cloud.google.com/bigquery/docs/reference/libraries)
  - [ ] - test
- Datastore
  - [x] - sample, [already published](https://cloud.google.com/datastore/docs/reference/libraries)
  - [ ] - test
- Natural Language
  - [x] - sample, [already published](https://cloud.google.com/natural-language/docs/reference/libraries)
  - [ ] - test
- Logging
  - [x] - sample, [already published](https://cloud.google.com/logging/docs/reference/libraries)
  - [ ] - test
- Monitoring
  - [x] - sample, [already published](https://cloud.google.com/monitoring/docs/reference/libraries)
  - [ ] - test
- Pub/Sub
  - [x] - sample, [already published](https://cloud.google.com/pubsub/docs/reference/libraries)
  - [ ] - test
- Speech
  - [x] - sample, [already published](https://cloud.google.com/speech/docs/reference/libraries)
  - [ ] - test
- Storage
  - [x] - sample, [already published](https://cloud.google.com/storage/docs/reference/libraries)
  - [ ] - test
- Translate
  - [x] - sample, [already published](https://cloud.google.com/translate/docs/reference/libraries)
  - [ ] - test
- Vision
  - [x] - sample
  - [ ] - test

For reference, see the corresponding samples in other languages:
- [GoogleCloudPlatform/nodejs-docs-samples](https://github.com/GoogleCloudPlatform/nodejs-docs-samples) (samples are on `master` in various `quickstart.js` files)
- GoogleCloudPlatform/python-docs-samples#547
- GoogleCloudPlatform/ruby-docs-samples#77
- GoogleCloudPlatform/php-docs-samples#174
- GoogleCloudPlatform/java-docs-samples (pending)
- GoogleCloudPlatform/dotnet-docs-samples (pending)

Thanks!

### CODE REVIEW

**Comment by broady** [comment]
I'm wondering why these snippets are to live in this repo and not as top-level godoc examples in the library code itself. This is specific to Go, and something we're having ongoing discussions with @rakyll and the eng team (including @jba).

A few high level comments:
1. Move each one into a `*_quickstart` directory. In Go, each package is a directory. The filename is not relevant. For example, the Vision quickstart might be in `vision/vision_quickstart`.
2. Add a package-level doc to each file. For example, vision would be `// Sample vision_quickstart is a ....`
3. Run `gofmt` over each file.
4. `projectId` should be `projectID`
5. There is no "instantiation" in Go. Use "Create". But more generally, these comments don't seem very useful. We can discuss that off-thread, though. I understand you put a lot of thought into it already.
6. `// Imports the Google Cloud client library` should be more specific, and not use the name "library". `// Imports the Google Cloud Datastore client package` or similar.
7. No need for region tags if the entire file is being included.

I will comment on specific code/files once the above is done.

**Comment by jmdobry** [comment]
> I'm wondering why these snippets are to live in this repo and not as top-level godoc examples in the library code itself. This is specific to Go, and something we're having ongoing discussions with @rakyll and the eng team (including @jba).

Whether you eventually want https://cloud.google.com to source samples from `google-cloud-go` is fine with me. It was just easier for me to submit the PR to this repo.

I'll work through the rest of your comments.

**Comment by broady** [comment]
Yeah, perfect. Don't worry too much about the directory name, then.

If you prefer, I (or Jaana) can also just take it from here and put it in the appropriate place, and add you as a reviewer.

**Comment by jmdobry** [comment]
> Move each one into a `*_quickstart` directory. In Go, each package is a directory. The filename is not relevant. For example, the Vision quickstart might be in `vision/vision_quickstart`.

Done

> Add a package-level doc to each file. For example, vision would be `// Sample vision_quickstart is a ....`

Done

> Run `gofmt` over each file.

Done

>  `projectId` should be `projectID`

Done

> There is no "instantiation" in Go. Use "Create". 

Done

> `// Imports the Google Cloud client library` should be more specific, and not use the name "library". `// Imports the Google Cloud Datastore client package` or similar.

Done

File: bigquery/bigquery_quickstart/main.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// [START bigquery_quickstart]
+// Sample bigquery_quickstart creates a Google BigQuery dataset.
+package main
+
+import (
+	"fmt"
+	"golang.org/x/net/context"
+	"log"
+
+	// Imports the Google Cloud Datastore client package
```
Comment (rakyll) [inline bigquery/bigquery_quickstart/main.go]: We follow the typical Go comment style to keep these types of guidance comments as proper sentences.

// Imports the Google Cloud Datastore client package.

File: bigquery/bigquery_quickstart/main.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// [START bigquery_quickstart]
+// Sample bigquery_quickstart creates a Google BigQuery dataset.
```
Comment (rakyll) [inline bigquery/bigquery_quickstart/main.go]: It is not very common to name packages with an underscore in Go. Can we call name these as bigquery-quickstart?

File: bigquery/bigquery_quickstart/main.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	// Creates a client
+	client, err := bigquery.NewClient(ctx, projectID)
+	if err != nil {
+		log.Fatalf("Failed to create client: %v", err)
+	}
+
+	// The name for the new dataset
+	datasetName := "my_new_dataset"
+
+	// Prepares the new dataset
```
Comment (rakyll) [inline bigquery/bigquery_quickstart/main.go]: Maybe we should just say "Create a dataset instance." if we need to explain the line.

File: bigquery/bigquery_quickstart/main.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	dataset := client.Dataset(datasetName)
+
+	// Creates the dataset
+	if err := dataset.Create(ctx); err != nil {
+		log.Fatalf("Failed to create dataset: %v", err)
+	}
+
+	fmt.Printf("Dataset %v created", dataset)
```
Comment (rakyll) [inline bigquery/bigquery_quickstart/main.go]: Dataset doesn't have a nice Stringer. I'd just leave this as fmt.Println("Dataset created.")

Comment (rakyll) [inline bigquery/bigquery_quickstart/main.go]: Missing new line at the end.

File: datastore/datastore_quickstart/main.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+	// Prepares the new entity
+	task := new(Task)
+	task.Description = "Buy milk"
+
+	// Saves the entity
+	if _, err := client.Put(ctx, taskKey, task); err != nil {
+		log.Fatalf("Failed to save task: %v", err)
+	}
+
+	fmt.Printf("Saved %v: %v", taskKey.String(), task.Description)
```
Comment (rakyll) [inline datastore/datastore_quickstart/main.go]: s/taskKey.String()/taskKey. fmt.Printf will automatically call the String. See https://play.golang.org/p/2ZykcW2pMH.

Missing new line at the end.

File: datastore/datastore_quickstart/main.go [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+	// The kind for the new entity
+	kind := "Task"
+	// The name/ID for the new entity
+	name := "sampletask1"
+	// The Cloud Datastore key for the new entity
+	taskKey := datastore.NewKey(ctx, kind, name, 0, nil)
+
+	// Prepares the new entity
+	task := new(Task)
```
Comment (rakyll) [inline datastore/datastore_quickstart/main.go]: Just use

```
task := Task {
   Description: "Buy milk",
}
```

File: datastore/datastore_quickstart/main.go [hunk 7]
```diff
@@ -0,0 +1,53 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// [START datastore_quickstart]
+// Sample datastore_quickstart fetches an entity from Google Cloud Datastore.
```
Comment (rakyll) [inline datastore/datastore_quickstart/main.go]: ditto the name

File: language/language_quickstart/main.go [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+// license that can be found in the LICENSE file.
+
+// [START language_quickstart]
+// Sample language_quickstart uses the Google Cloud Natural API to analyze the
+// sentiment of "Hello, world!".
+package main
+
+import (
+	"fmt"
+	"golang.org/x/net/context"
```
Comment (rakyll) [inline language/language_quickstart/main.go]: Move the import to the next group.

The top group belongs to the standard library imports.

File: language/language_quickstart/main.go [hunk 9]
```diff
@@ -0,0 +1,54 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// [START language_quickstart]
+// Sample language_quickstart uses the Google Cloud Natural API to analyze the
```
Comment (rakyll) [inline language/language_quickstart/main.go]: ditto for the name

File: datastore/datastore_quickstart/main.go [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+package main
+
+import (
+	"fmt"
+	"golang.org/x/net/context"
```
Comment (rakyll) [inline datastore/datastore_quickstart/main.go]: Move the context import to the next group. The uppermost group belongs to the standard library imports.

File: bigquery/bigquery_quickstart/main.go [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// [START bigquery_quickstart]
+// Sample bigquery_quickstart creates a Google BigQuery dataset.
+package main
+
+import (
+	"fmt"
+	"golang.org/x/net/context"
```
Comment (rakyll) [inline bigquery/bigquery_quickstart/main.go]: Move the context import to the next group. The uppermost group belongs to the standard library imports.

File: language/language_quickstart/main.go [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+		},
+		EncodingType: languagepb.EncodingType_UTF8,
+	})
+	if err != nil {
+		log.Fatalf("Failed to analyze text: %v", err)
+	}
+
+	fmt.Printf("Text: %v\n", text)
+	if sentiment.DocumentSentiment.Score >= 0 {
+		fmt.Printf("Sentiment: positive")
```
Comment (rakyll) [inline language/language_quickstart/main.go]: fmt.Println instead.

File: language/language_quickstart/main.go [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+	} else {
+		fmt.Printf("Sentiment: negative")
```
Comment (rakyll) [inline language/language_quickstart/main.go]: fmt.Println instead.

File: logging/logging_quickstart/main.go [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// [START logging_quickstart]
+// Sample logging_quickstart writes a log entry to Stackdriver Logging.
+package main
+
+import (
+	"fmt"
+	"golang.org/x/net/context"
```
Comment (rakyll) [inline logging/logging_quickstart/main.go]: ditto for the context import

File: logging/logging_quickstart/main.go [hunk 15]
```diff
@@ -0,0 +1,51 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// [START logging_quickstart]
+// Sample logging_quickstart writes a log entry to Stackdriver Logging.
```
Comment (rakyll) [inline logging/logging_quickstart/main.go]: ditto for the name

File: logging/logging_quickstart/main.go [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+	logger := client.Logger(logName)
+
+	// The data to log
+	text := "Hello, world!"
+
+	// Adds an entry to the log buffer
+	logger.Log(logging.Entry{Payload: text})
+
+	// Closes the client and flushes the buffer to the Stackdriver Logging service
+	err = client.Close()
```
Comment (rakyll) [inline logging/logging_quickstart/main.go]: if err := client.Close(); err != nil

File: logging/logging_quickstart/main.go [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+	if err != nil {
+		log.Fatalf("Failed to close client: %v", err)
+	}
+
+	fmt.Printf("Logged: %v", text)
```
Comment (rakyll) [inline logging/logging_quickstart/main.go]: missing new line

File: pubsub/pubsub_quickstart/main.go [hunk 18]
```diff
@@ -0,0 +1,42 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// [START pubsub_quickstart]
+// Sample pubsub_quickstart creates a Google Cloud Pub/Sub topic.
```
Comment (rakyll) [inline pubsub/pubsub_quickstart/main.go]: ditto for the name

File: pubsub/pubsub_quickstart/main.go [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+package main
+
+import (
+	"fmt"
+	"golang.org/x/net/context"
```
Comment (rakyll) [inline pubsub/pubsub_quickstart/main.go]: ditto for the import group

File: pubsub/pubsub_quickstart/main.go [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+	// The name for the new topic
+	topicName := "my-new-topic"
+
+	// Creates the new topic
+	topic, err := client.CreateTopic(ctx, topicName)
+	if err != nil {
+		log.Fatalf("Failed to create topic: %v", err)
+	}
+
+	fmt.Printf("Topic %v created.", topic)
```
Comment (rakyll) [inline pubsub/pubsub_quickstart/main.go]: Missing new line

File: storage/storage_quickstart/main.go [hunk 21]
```diff
@@ -0,0 +1,44 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// [START storage_quickstart]
+// Sample storage_quickstart creates a Google Cloud Storage bucket.
```
Comment (rakyll) [inline storage/storage_quickstart/main.go]: ditto name

File: storage/storage_quickstart/main.go [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+package main
+
+import (
+	"fmt"
+	"golang.org/x/net/context"
```
Comment (rakyll) [inline storage/storage_quickstart/main.go]: ditto import group

File: translate/translate_quickstart/main.go [hunk 23]
```diff
@@ -0,0 +1,46 @@
...
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// [START translate_quickstart]
+// Sample translate_quickstart translates "Hello, world!" into Russian.
+package main
+
+import (
+	"fmt"
+	"golang.org/x/net/context"
```
Comment (rakyll) [inline translate/translate_quickstart/main.go]: Move "golang.org/x/net/context" and "golang.org/x/net/context" to another group.

File: translate/translate_quickstart/main.go [hunk 24]
```diff
@@ -0,0 +1,46 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// [START translate_quickstart]
```
Comment (rakyll) [inline translate/translate_quickstart/main.go]: ditto fro the name

File: translate/translate_quickstart/main.go [hunk 25]
```diff
@@ -0,0 +1,46 @@
...
+	}
+
+	// Translates some text into Russian
+	translations, err := client.Translate(ctx, []string{text}, target, nil)
+	if err != nil {
+		log.Fatalf("Failed to translate text: %v", err)
+	}
+
+	fmt.Printf("Text: %v\n", text)
+	fmt.Printf("Translation: %v", translations[0].Text)
```
Comment (rakyll) [inline translate/translate_quickstart/main.go]: Missing new line

File: vision/vision_quickstart/main.go [hunk 26]
```diff
@@ -0,0 +1,49 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// [START vision_quickstart]
```
Comment (rakyll) [inline vision/vision_quickstart/main.go]: ditto fofr thte name

File: vision/vision_quickstart/main.go [hunk 27]
```diff
@@ -0,0 +1,49 @@
...
+// Sample vision_quickstart uses the Google Cloud Vision API to label an image.
+package main
+
+import (
+	"fmt"
+	"golang.org/x/net/context"
```
Comment (rakyll) [inline vision/vision_quickstart/main.go]: move context to another group.

File: vision/vision_quickstart/main.go [hunk 28]
```diff
@@ -0,0 +1,49 @@
...
+	ctx := context.Background()
+
+	// Creates a client
+	client, err := vision.NewClient(ctx)
+	if err != nil {
+		log.Fatalf("Failed to create client: %v", err)
+	}
+
+	// The name of the image file to annotate
+	fileName := "vision/testdata/cat.jpg"
```
Comment (rakyll) [inline vision/vision_quickstart/main.go]: s/fileName/filename. See https://en.wikipedia.org/wiki/Filename.

File: vision/vision_quickstart/main.go [hunk 29]
```diff
@@ -0,0 +1,49 @@
...
+	if err != nil {
+		log.Fatalf("Failed to read file: %v", err)
+	}
+	defer file.Close()
+	image, err := vision.NewImageFromReader(file)
+	if err != nil {
+		log.Fatalf("Failed to create image: %v", err)
+	}
+
+	labels, err := client.DetectLabels(ctx, image, 10)
```
Comment (rakyll) [inline vision/vision_quickstart/main.go]: Missing error check.

File: vision/vision_quickstart/main.go [hunk 30]
```diff
@@ -0,0 +1,49 @@
...
+
+	fmt.Printf("Labels:\n")
```
Comment (rakyll) [inline vision/vision_quickstart/main.go]: or fmt.Println("Labels:")

File: bigquery/bigquery_quickstart/main.go (hunk 2 above)
Comment (jmdobry) [inline bigquery/bigquery_quickstart/main.go]: Done

File: bigquery/bigquery_quickstart/main.go (hunk 11 above)
Comment (jmdobry) [inline bigquery/bigquery_quickstart/main.go]: Done

File: bigquery/bigquery_quickstart/main.go (hunk 1 above)
Comment (jmdobry) [inline bigquery/bigquery_quickstart/main.go]: Done

File: bigquery/bigquery_quickstart/main.go (hunk 3 above)
Comment (jmdobry) [inline bigquery/bigquery_quickstart/main.go]: Done

File: bigquery/bigquery_quickstart/main.go (hunk 4 above)
Comment (jmdobry) [inline bigquery/bigquery_quickstart/main.go]: Done

Comment (jmdobry) [inline bigquery/bigquery_quickstart/main.go]: Done

File: datastore/datastore_quickstart/main.go (hunk 7 above)
Comment (jmdobry) [inline datastore/datastore_quickstart/main.go]: Done

File: datastore/datastore_quickstart/main.go (hunk 10 above)
Comment (jmdobry) [inline datastore/datastore_quickstart/main.go]: Done

File: datastore/datastore_quickstart/main.go (hunk 6 above)
Comment (jmdobry) [inline datastore/datastore_quickstart/main.go]: Done

File: datastore/datastore_quickstart/main.go (hunk 5 above)
Comment (jmdobry) [inline datastore/datastore_quickstart/main.go]: Done

File: language/language_quickstart/main.go (hunk 9 above)
Comment (jmdobry) [inline language/language_quickstart/main.go]: Done

File: language/language_quickstart/main.go (hunk 8 above)
Comment (jmdobry) [inline language/language_quickstart/main.go]: Done

File: language/language_quickstart/main.go (hunk 12 above)
Comment (jmdobry) [inline language/language_quickstart/main.go]: Done

File: language/language_quickstart/main.go (hunk 13 above)
Comment (jmdobry) [inline language/language_quickstart/main.go]: Done

File: logging/logging_quickstart/main.go (hunk 15 above)
Comment (jmdobry) [inline logging/logging_quickstart/main.go]: Done

File: logging/logging_quickstart/main.go (hunk 14 above)
Comment (jmdobry) [inline logging/logging_quickstart/main.go]: Done

File: logging/logging_quickstart/main.go (hunk 16 above)
Comment (jmdobry) [inline logging/logging_quickstart/main.go]: Done

File: logging/logging_quickstart/main.go (hunk 17 above)
Comment (jmdobry) [inline logging/logging_quickstart/main.go]: Done

File: pubsub/pubsub_quickstart/main.go (hunk 18 above)
Comment (jmdobry) [inline pubsub/pubsub_quickstart/main.go]: Done

File: pubsub/pubsub_quickstart/main.go (hunk 19 above)
Comment (jmdobry) [inline pubsub/pubsub_quickstart/main.go]: Done

File: pubsub/pubsub_quickstart/main.go (hunk 20 above)
Comment (jmdobry) [inline pubsub/pubsub_quickstart/main.go]: Done

File: storage/storage_quickstart/main.go (hunk 21 above)
Comment (jmdobry) [inline storage/storage_quickstart/main.go]: Done

File: storage/storage_quickstart/main.go (hunk 22 above)
Comment (jmdobry) [inline storage/storage_quickstart/main.go]: Done

File: translate/translate_quickstart/main.go (hunk 24 above)
Comment (jmdobry) [inline translate/translate_quickstart/main.go]: Done

File: translate/translate_quickstart/main.go (hunk 23 above)
Comment (jmdobry) [inline translate/translate_quickstart/main.go]: Done

File: translate/translate_quickstart/main.go (hunk 25 above)
Comment (jmdobry) [inline translate/translate_quickstart/main.go]: Done

File: vision/vision_quickstart/main.go (hunk 26 above)
Comment (jmdobry) [inline vision/vision_quickstart/main.go]: Done

File: vision/vision_quickstart/main.go (hunk 27 above)
Comment (jmdobry) [inline vision/vision_quickstart/main.go]: Done

File: vision/vision_quickstart/main.go (hunk 28 above)
Comment (jmdobry) [inline vision/vision_quickstart/main.go]: Done

File: vision/vision_quickstart/main.go (hunk 29 above)
Comment (jmdobry) [inline vision/vision_quickstart/main.go]: Done

File: vision/vision_quickstart/main.go (hunk 30 above)
Comment (jmdobry) [inline vision/vision_quickstart/main.go]: Done

**Comment by waprin** [comment]
@cbro @rakyll I want to get monitoring quickstart merged, and I want to get snippets written to check off checklist and make Omar happy. Happy to write them, just not sure what best course of action is here given this PR does tons of unrelated quickstarts. And I wanted to make sure you were on board with writing the snippets in this repo, seems you were ambivalent? We need to include them on devsite so that's major source of motivation, no idea if autogen VTK samples are sufficient but doubtful).

**Comment by broady** [comment]
@waprin moving to e-mail

**Pull request merged**